src/serializers/type_serializers/bytes.rs

```diff
@@ -55,9 +55,14 @@ impl TypeSerializer for BytesSerializer {
         }
     }
 
-    fn json_key<'a>(&self, key: &'a Bound<'_, PyAny>, extra: &Extra) -> PyResult<Cow<'a, str>> {
+    fn json_key<'py>(&self, key: &Bound<'py, PyAny>, extra: &Extra) -> PyResult<Cow<'py, str>> {
```

Comment:
@davidhewitt I change the lifetime signature of `json_key` to a value I think is more correct, it worked for the new case in `enum_.rs`, but it broke lifetimes in this one function.

My simplistic assumption is that `PyBytes.as_bytes()` is wrongly saying the data has lifetime of the reference, not `'py` but I could be wrong. Either way I'd appreciate your review.

Reply:
I've played around with this a bit more, and it makes a little more sense. If there isn't a way to square this circle, I think the best solution would be to revert to the old lifetimes and just clone the string in `enum_.rs` which will be a rare case.

Reply:
The bytes data is indeed limited by the reference and not by `'py`.

> revert to the old lifetimes and just clone the string in `enum_.rs` which will be a rare case

Yes I think that's the right approach. PyO3 0.21 also has new types `PyBackedStr` and `PyBackedBytes` which are like `&str` and `&[u8]` but without lifetimes, so maybe these can help across the board to avoid copying.

Reply:
fixed.